### DISCUSSION TITLE
GSoC 2024 Projects

### DISCUSSION THREAD

**Discussion by MatthewMiddlehurst** [aeon-toolkit/aeon#1121] [Ideas]
This discussion is for the `aeon` [Google Summer of Code (GSoC) 2024](https://summerofcode.withgoogle.com/) project proposals. 

Feel free to ask any questions regarding the projects and GSoC, or just introduce yourself and say hello! Questions and general `aeon` discussion can also be sent on the projects [Slack](https://join.slack.com/t/aeon-toolkit/shared_invite/zt-22vwvut29-HDpCu~7VBUozyfL_8j3dLA).

The link to our list of projects for GSoC is [here](https://github.com/aeon-toolkit/aeon-admin/blob/main/gsoc/gsoc-2024-projects.md). We are applying for GSoC under the [NumFOCUS](https://github.com/numfocus/gsoc) umbrella.

If you wish to contribute to `aeon`, please take a look at our [contributing guide](https://www.aeon-toolkit.org/en/latest/contributing.html).

Applications for GSoC open March 18th and close April 2nd.

**Comment by Paulnkk** [comment]
Hey @MatthewMiddlehurst, 

thanks for opening this discussion! I am Paul and currently phd student at KIT in Germany (Electrical Engineering and operations research). I'm really excited about the Machine Learning for Time Series Forecasting project. In my research I am currently working a lot with decision tree models (especially MIP formulations) and have already had experience with time series analysis due to my experience in finance Is it possible to exchange information about the project via email or to see whether there is a fit with this project proposal? I think also working on a paper in this context would be a great opportunity!

Best regards

Paul

**Reply by TonyBagnall** [reply]
hi Paul, that sounds great, feel free to email us on aeon-toolkit@gmail.com, or email me directly (A.J.Bagnall@soton.ac.uk) to discuss

**Reply by MatthewMiddlehurst** [reply]
Hi Paul, 

Glad to hear you are interested! There is definitely scope for writing something up depending on the results of the project and how far we get. We have a project-wide email at contact@aeon-toolkit.org, feel free to ask questions there or continue here. The mentors may have emails on their GitHub page linked in the proposals also.

Matthew

**Comment by Paulnkk** [comment]
@TonyBagnall @MatthewMiddlehurst Thanks a lot for the fast response! I will get in contact with you via mail

**Comment by itsdivya1309** [comment]
Hii @MatthewMiddlehurst !
I am Divya Tiwari, an undergraduate engineering student at IIT (BHU) Varanasi, India. I've been working with aeon for some time and am interested in the classification part of Time Series Analysis. I have experience in Python OOPs and Machine Learning. I am interested in the project **Implement Proximity Forest for Time Series Classification** and would like to work on it.

**Reply by MatthewMiddlehurst** [reply]
Thanks for your interest and making a PR! We are not sure if our projects have been accepted yet, but are hopeful 🙂. Feel free to ask any questions on here or on Slack. We will update when we know more.

**Reply by itsdivya1309** [reply]
Thank you. I am currently reading the papers for the project. I will update you once I'm done.

**Comment by lfun1** [comment]
Hi @MatthewMiddlehurst,

I'm Lisa Fung, a CS + Math undergraduate at Stanford University. I have experience working with tensor decompositions for GIS, and I'm looking to integrate advanced time series analysis in my research. I have a strong grasp of OOP, machine learning, and data science, and I would love to work on **Project 2: Improving the computational complexity of the Shapelet Transform** this summer.

**Reply by baraline** [reply]
Hi Lisa, glad to hear that you are interested in the project ! If you have any question met us know.

**Comment by Alex-osayi** [comment]
Hi @MatthewMiddlehurst 
I am Alexander Osadolor, I just concluded a Data Science bootcamp at HyperionDev, and I am currently pursuing a master’s degree at Teesside University. I find "Machine Learning for Time Series Forecasting" intriguing, and would gladly love to be a part of it.
Regards

**Reply by MatthewMiddlehurst** [reply]
Hi, good to hear! There is still time to create an application. Best of luck, let us know if you have any questions.

**Reply by ganeshgaiy** [reply]
Hey @MatthewMiddlehurst

Can you please guide me on how to create an application? Hope I am not too late for this!

**Reply by MatthewMiddlehurst** [reply]
@ganeshgaiy you would have to be very quick, I think it may be a bit late at this point. You can follow the guidance here.

You make your application on the GSoC webpage: https://summerofcode.withgoogle.com/.
There is some guidance from NumFOCUS here: https://github.com/numfocus/gsoc/blob/master/CONTRIBUTING-students.md.

**Reply by ganeshgaiy** [reply]
Thank you so much for the reply @MatthewMiddlehurst. I was unable to send an application. Can I still contribute? My main goal was to receive mentorship and increase my ML skills.

**Reply by MatthewMiddlehurst** [reply]
Yes, feel free to contribute code or documentation as you like. Visit our webpage for a contributors guide.
